Wikipedia/Code/MWKDataStore.m

```diff
@@ -1664,7 +1671,12 @@ - (void)updateLocalConfigurationFromRemoteConfigurationWithCompletion:(nullable
                                             [taskGroup leave];
                                             return;
                                         }
-                                        [self updateLocalConfigurationFromRemoteConfiguration:remoteConfigurationDictionary];
+                                        if (self.isLocalConfigUpdateAllowed) {
+                                            [self updateLocalConfigurationFromRemoteConfiguration:remoteConfigurationDictionary];
+                                            self.failedToUpdateLocalConfig = NO;
+                                        } else {
+                                            self.failedToUpdateLocalConfig = YES;
```

Comment:
@nambatee I feel like maybe you should keep track of separate failed flags here from the updateReadingListsLimits call above. Is there a chance that completion could flip failedToUpdateLocalConfig again before it's checked?

Reply:
Good point, it can definitely flip